src/app/shared/cookies/browser-klaro.service.spec.ts

```diff
@@ -233,4 +260,46 @@ describe('BrowserKlaroService', () => {
       expect(ePersonService.patch).not.toHaveBeenCalled();
     });
   });
+
+  describe('initialize google analytics configuration', () => {
+    beforeEach(() => {
+      configurationDataService.findByPropertyName = findByPropertyName;
+      spyOn((service as any), 'getUser$').and.returnValue(observableOf(user));
+      translateService.get.and.returnValue(observableOf('loading...'));
+      spyOn(service, 'addAppMessages');
+      spyOn((service as any), 'initializeUser');
+      spyOn(service, 'translateConfiguration');
+    });
+    it('should have been initialized with googleAnalytics', () => {
+      const filteredServices = googleAnalyticsFilter(service);
+      expect(filteredServices.length).toBe(1);
+    });
+    it('should filter empty configuration', () => {
+      configurationDataService.findByPropertyName = jasmine.createSpy().and.returnValue(
+        createSuccessfulRemoteDataObject$({
+          ... new ConfigurationProperty(),
+          name: googleAnalytics,
+          values: [],
+        }));
+      service.initialize();
+      const filteredServices = googleAnalyticsFilter(service);
```

Comment:
In the context of testing `initialize()` it's a bit problematic that you don't assert that the filtering method was called (or that it filtered anything).

Please replace this local `googleAnalyticsFilter` function with a spy on the actual method.